### PR TITLE
Add support for Monolog3

### PR DESCRIPTION
**Pull request by morawskim**
This PR closes issue #26.

Monolog3 made some changes in API.

> Log records have been converted from an array to a Monolog\LogRecord object with public (and mostly readonly) properties. e.g. instead of doing $record['context'] use $record->context. In formatters or handlers if you rather need an array to work with you can use $record->toArray() to get back a Monolog 1/2 style record array. This will contain the enum values instead of enum cases in the level and level_name keys to be more backwards compatible and use simpler data types.

This PR fix error:
```
Chrisguitarguy\RequestId\Monolog\RequestIdProcessor::__invoke(): Argument #1 ($record) must be of type array, Monolog\LogRecord given, called in /app/vendor/monolog/monolog/src/Monolog/Logger.php on line 331
```

Monolog3 requires PHP 8.1.

You can use my demo - https://github.com/morawskim/RequestIdBundle-test
Or use instruction from issue #26.

Fetch a git repository `git clone https://github.com/morawskim/RequestIdBundle-test.git`
Call `docker-compose up -d`
Install dependencies - `docker-compose exec php composer install`

Open web browser and go to `localhost:8080`
You should see a error message:
```
Chrisguitarguy\RequestId\Monolog\RequestIdProcessor::__invoke(): Argument #1 ($record) must be of type array, Monolog\LogRecord given, called in /app/vendor/monolog/monolog/src/Monolog/Logger.php on line 331
```

Switch branch to `fork`.
Install dependencies - `docker-compose exec php composer install`
Refresh webpage. You should see a default Symfony application homepage.

### CODE REVIEW

**Review by chrisguitarguy — APPROVED** [review]

**Comment by morawskim** [comment]
I have added the symfony/flex to composer allow-plugins to fix the pipeline

**Comment by morawskim** [comment]
Hello  @chrisguitarguy 

Are we waiting for something before merge this PR?
Can I help?

**Comment by chrisguitarguy** [comment]
I honestly just forgot about it 😬 I will merge it now

**Comment by chrisguitarguy** [comment]
Tagged v5.1.0 with this too

**Pull request merged**
